gammapy/irf/core.py

```diff
@@ -201,8 +201,30 @@ def evaluate(self, method=None, **kwargs):
             coord = kwargs.get(key, value)
             if coord is not None:
                 coords_default[key] = u.Quantity(coord, copy=False)
-
-        return self._interpolate(coords_default.values(), method=method)
+        data = self._interpolate(coords_default.values(), method=method)
+
+        eps = 1e-5
+        for k, ax in enumerate(self.axes):
+            coords = coords_default[ax.name]
+            ndim = len(np.atleast_1d(coords))
+            unit = coords.unit
+            if ndim == 0:
+                continue
+            if ndim > 1:
+                coords = coords.squeeze()
+            if coords.shape != data.shape:
+                axis = tuple(
+                    [idx for idx in range(len(data.shape)) if idx != k]
+                )  # simpler solution ?
+                coords = np.expand_dims(coords, axis=axis)
+                coords = np.broadcast_to(coords, data.shape) * unit
+            axmin = ax.edges.min()
+            axmax = ax.edges.max()
+            mask = (coords < (1 - eps * np.sign(axmin)) * axmin) | (
+                coords > (1 + eps * np.sign(axmax)) * axmax
+            )
+            data[mask] = np.nan
+        return data
```

Comment:
I tried to add this code to allow the extrapolation within the default range, it fix few tests but I still have problems with array shapes in many cases. @adonath a suggestion to make this simpler ?

Reply:
I think something along the lines of:
```python3
from gammapy.maps import MapAxis, MapAxes
from astropy import units as u
import numpy as np

energy = MapAxis.from_energy_bounds("1 TeV", "10 TeV", nbin=10)
offset = MapAxis.from_bounds(0, 5, unit="deg", nbin=5, name="offset")

axes = MapAxes([energy, offset])
coords = {
    "energy": np.geomspace(0.3, 30, 20)[:, None] * u.TeV,
    "offset": np.linspace(0, 6, 10) * u.deg
}

idxs = axes.coord_to_idx(
    coords, clip=False
)

invalid = np.broadcast_arrays(*[idx == -1 for idx in idxs])
mask = np.any(invalid, axis=0)
```
Should work. I think the correct clipping of values should be handled in `MapAxis.coord_to_idx()` if to doesn't work as expected we should probably adapt it there.

Reply:
I haven't looked into your solution in detail, but `._interpolate` does the broadcasting of arrays internally as well, so `data` and `mask` should always end up with the same shape.

Reply:
Thanks this is much simple. The evaluation at the edge value sometimes return nan,  I think we need have to add some tolerance, can we handle this doing the clipping in MapAxis.coord_to_idx() ? Otherwise we could just adapt the tests.

Reply:
Do you have a minimal test case for this, or which test?